dpnp/dpnp_iface_manipulation.py

```diff
@@ -183,74 +189,122 @@ def atleast_2d(*arys):
+    array([[0., 1., 2.]])
+
     """
 
-    all_is_array = True
-    arys_desc = []
+    res = []
     for ary in arys:
-        if not dpnp.isscalar(ary):
-            ary_desc = dpnp.get_dpnp_descriptor(
-                ary, copy_when_nondefault_queue=False
+        if not dpnp.is_supported_array_type(ary):
+            raise TypeError(
+                "All arrays must be any of supported type, "
+                f"but got {type(ary)}"
             )
-            if ary_desc:
-                arys_desc.append(ary_desc)
-                continue
-        all_is_array = False
-        break
-
-    if not use_origin_backend(arys[0]) and all_is_array:
-        result = []
-        for ary_desc in arys_desc:
-            res = dpnp_atleast_2d(ary_desc).get_pyobj()
-            result.append(res)
-
-        if len(result) == 1:
-            return result[0]
+        if ary.ndim == 0:
+            result = ary.reshape(1, 1)
+        elif ary.ndim == 1:
+            result = ary[dpnp.newaxis, :]
         else:
-            return result
-
-    return call_origin(numpy.atleast_2d, *arys)
+            result = ary
+        if isinstance(result, dpt.usm_ndarray):
+            result = dpnp_array._create_from_usm_ndarray(result)
+        res.append(result)
+    if len(res) == 1:
+        return res[0]
+    else:
+        return res
 
 
-def atleast_3d(*arys):
+def atleast_3d(*arys, device=None, usm_type=None, sycl_queue=None):
     """
     View inputs as arrays with at least three dimensions.
 
     For full documentation refer to :obj:`numpy.atleast_3d`.
 
-    Limitations
-    -----------
-    Input arrays is supported as :obj:`dpnp.ndarray`.
+    Parameters
+    ----------
+    arys : {dpnp.ndarray, usm_ndarray}
+        One or more array-like sequences. Arrays that already have three or more
+        dimensions are preserved.
+
+    Returns
+    -------
+    out : dpnp.ndarray
+        An array, or list of arrays, each with ``a.ndim >= 3``. Copies are
+        avoided where possible, and views with three or more dimensions are
+        returned.
+
+    See Also
+    --------
+    :obj:`dpnp.atleast_1d` : Convert inputs to arrays with at least one dimension.
+    :obj:`dpnp.atleast_2d` : View inputs as arrays with at least three dimensions.
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> x = np.array(3.0)
+    >>> np.atleast_3d(x)
+    array([[[3.]]])
+
+    >>> x = np.arange(3.0)
+    >>> np.atleast_3d(x).shape
+    (1, 3, 1)
+
+    >>> x = np.arange(12.0).reshape(4, 3)
+    >>> np.atleast_3d(x).shape
+    (4, 3, 1)
+
     """
 
-    all_is_array = True
-    arys_desc = []
+    res = []
     for ary in arys:
-        if not dpnp.isscalar(ary):
-            ary_desc = dpnp.get_dpnp_descriptor(
-                ary, copy_when_nondefault_queue=False
+        if not dpnp.is_supported_array_type(ary):
+            raise TypeError(
+                "All arrays must be any of supported type, "
+                f"but got {type(ary)}"
```

Comment:
here and for others:
```suggestion
                "Each input array must be any of supported type, "
                f"but got {type(ary)}"
```